### DISCUSSION TITLE
FAST.farm problems

### DISCUSSION THREAD

**Discussion by YeraiP** [OpenFAST/openfast#1660] [Q&A]
Hi, 

I'm trying to simulate the three wind turbine farm example case provided with fast.FARM (the one with three of turbines in the same row, one behind the other) for four different wind speeds (8, 12, 16, and 20mps) using FAST farm. I generate the four wind files using turbsim and then, I run FAST.farm from matlab for the four different winds. The results for the first three winds seem fine, but when I simulate any wind faster than around 18mps the results I get are wrong. I usually get some of the turbines fine, but the rest give some results that are like a chattering signal. See for example a figure with the generator speed for the three turbines below, where the results from the second turbine are wrong.

![example](https://github.com/OpenFAST/openfast/assets/40027163/1ffd82c3-9c1e-4af4-b5bf-8c3929a65d4d)

I should clarify, that the results for all the variables (but the wind speed) look similar, with the same chattering, that depending on the simulation the wrong results happen on a different turbine, and that I run all the codes from the same Matlab script, so it shouldn't be an error there, because in that case all of them would give the same error. I use the precompiled FAST.Farm_v3.2.1_x64.exe executable.

Did anybody have this issue before or know how I could solve it?

Thank you very much in advance,
Yerai

**Comment by jjonkman** [comment]
Dear @YeraiP,

I'm not sure I know enough about what you are doing to pinpoint the issue.  Just a couple comments:

- Have you set proper initial conditions of rotor speed and blade-pitch angles (based on their expected mean value for the given hub-height wind speed you are simulating) for each mean wind speed you are simulating?
- Are you following the FAST.Farm modeling guidance for defining the spatial-temporal resolution of the low- and high-resolution domains, some of which are dependent on mean wind speed?  In general, I would recommend using the scripts available in the python toolbox when setting up FAST.Farm simulations for different conditions (https://github.com/OpenFAST/python-toolbox/tree/main/pyFAST/fastfarm), rather than relying on the r-test examples, which are there more for benchmarking as FAST.Farm is updated rather than as good examples that should be replicated for different analyses.

Best regards,

**Reply by YeraiP** [reply]
Dear Dr. Jonkman,

It was a problem of initial conditions, that I had the rotor speed in 12.0 instead of 12.1, and that solved it!

thank you!
Yerai

**Comment by mohsamaz** [comment]
Dear @jjonkman,

When running simulations in OpenFAST, I generate the wind field using TurbSim, which means the wind direction remains constant throughout the simulation.


I would like to know if FAST.Farm allows the wind direction to vary with time. Specifically, I have a time series of measured wind direction data from my wind farm, and I would like to reproduce this varying wind direction within FAST.Farm.

Is it possible to input a time-varying wind direction in FAST.Farm, and if so, what is the recommended approach?

Best regards,

**Reply by jjonkman** [reply]
Dear @mohsamaz,

See a similar question asked and answered on our forum: https://forums.nrel.gov/t/dynamic-wind-direction-variation-in-fast-farm-using-turbsim-or-user-defined-time-series/8365.

Best regards,
